target/linux/ar71xx/files/arch/mips/ath79/mach-rb91x.c

```diff
@@ -340,4 +335,4 @@ static void __init rb711gr100_setup(void)
 
 }
 
-MIPS_MACHINE_NONAME(ATH79_MACH_RB_711GR100, "711Gr100", rb711gr100_setup);
+MIPS_MACHINE(ATH79_MACH_RB_711GR100, "711Gr100", "MikroTik RouterBOARD 912UAG-5HPnD", rb711gr100_setup);
```

Comment:
This is wrong: it break all other RB devices that relied on that code (e.g. RB911G-*) which will now be mis-identified.
See https://github.com/openwrt/openwrt/blob/494babf34c7a41cf1a1775530ba0d2e4ebc0c43f/target/linux/ar71xx/files/arch/mips/ath79/mach-rb91x.c#L264

Reply:
Yes, sure, you are right, but first of all I wanna know if the trouble is in wrong mach detection of new revision, or somewhere else. I am not an expert of openwrt code and unfortunately I don't have the 91x new revision hardware to test myself. If this fix will help, then it could be possible to rewrite it to detect all other models.

Reply:
The problem is this is not a fix. What it does is statically declare that all machines booting with "711Gr100" board model will be detected and registered as "MikroTik RouterBOARD 912UAG-5HPnD". There's no way to "rewrite it to detect all other models". The previous code did exactly that.

The original code pulled the actual device name from the flash. As far as I know, all mikrotik devices store their model name in flash (in hard_config), it would be interesting to find out why this doesn't work with yours. The dumps provided by @xback above do contain that data.

HTH

Reply:
Thanks alot, I have compared 951 and 912 hard_configs and I think I have found my error in routerboard.c file in detection in rb_get_board_name function, I'll correct it at Monday. Thanks! You where right, there is no need to change mach-rb91x file.